packages/gitmoji-changelog-core/src/index.js

```diff
@@ -75,94 +75,106 @@ async function generateVersion(options) {
       }))
   }
 
-  return {
+  const result = {
     version,
-    date: version !== 'next' ? getLastCommitDate(commits) : undefined,
     groups: makeGroups(commits),
   }
-}
 
-async function generateVersions({ tags, groupSimilarCommits, meta }) {
-  let nextTag = ''
-
-  const initialFrom = meta && meta.lastVersion ? meta.lastVersion : ''
-  return Promise.all(
-    [...tags, initialFrom]
-      .map(tag => {
-        const params = {
-          groupSimilarCommits,
-          from: tag,
-          to: nextTag,
-          version: nextTag ? sanitizeVersion(nextTag) : 'next',
-        }
-        nextTag = tag
-        return params
-      })
-      .map(generateVersion)
-  )
-    .then(versions => versions.sort((c1, c2) => {
-      if (c1.version === 'next') return -1
-      if (c2.version === 'next') return 1
-      return semverCompare(c2.version, c1.version)
-    }))
-}
+  if (version !== 'next') {
+    result.date = getLastCommitDate(commits)
+  }
 
+  return result
+}
 async function generateChangelog(options = {}) {
   const {
     mode,
     release,
     groupSimilarCommits,
-    meta,
   } = options
 
-  const packageInfo = await getPackageInfo()
+  if (mode === 'init') {
+    const packageInfo = await getPackageInfo()
+    let requestedVersion = release === 'from-package' ? packageInfo.version : release
 
-  let version = release === 'from-package' ? packageInfo.version : release
-  if (version && version !== 'next') {
-    if (!semver.valid(version)) {
-      throw new Error(`${version} is not a valid semver version.`)
-    }
+    let tags = [...await gitSemverTagsAsync(), '']
```

Comment:
What is the meaning of this empty string ? Is this HEAD ? If yes, it's strange to have HEAD in a tag list. Could we add it later when it's useful ?

Reply:
This is the start. In case of an init we want to process everything 😄